sdk/keyvault/Azure.Security.KeyVault.Keys/src/Cryptography/Client/CryptographyClient.cs

```diff
@@ -0,0 +1,213 @@
+        /// <param name="credential">Represents a credential capable of providing an OAuth token.</param>
+        public CryptographyClient(Uri keyVaultKeyUri, TokenCredential credential)
+            : this(keyVaultKeyUri, credential, new CryptographyClientOptions()) { }
+
+        /// <summary>
+        /// Initializes a new instance of Cryptography Client
+        /// </summary>
+        /// <param name="keyVaultKeyUri">Endpoint URL for the Azure Key Vault service.</param>
+        /// <param name="credential">Represents a credential capable of providing an OAuth token.</param>
+        /// <param name="options">Options that allow to configure the management of the request sent to Key Vault.</param>
+        public CryptographyClient(Uri keyVaultKeyUri, TokenCredential credential, CryptographyClientOptions options) : this()
+        {
+            Check.NotNull(keyVaultKeyUri, nameof(keyVaultKeyUri));
+            Check.NotNull(options, nameof(options));
+            Check.NotNull(credential, nameof(credential));
+
+            KeyVaultKeyUri = keyVaultKeyUri;
+            CryptoClientOptions = options;
+            Credentials = credential;
+            BearerTokenAuthenticationPolicy bearerAuthPolicy = new BearerTokenAuthenticationPolicy(credential, DEFAULT_KV_SCOPE_URI);
+            Pipeline = HttpPipelineBuilder.Build(CryptoClientOptions, bufferResponse: true, clientPolicies: bearerAuthPolicy);
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="keyVaultKey"></param>
+        /// <param name="credential"></param>
+        public CryptographyClient(Key keyVaultKey, TokenCredential credential)
+            : this(keyVaultKey, credential, new CryptographyClientOptions()) { }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="keyVaultKey"></param>
+        /// <param name="credential"></param>
+        /// <param name="options"></param>
+        public CryptographyClient(Key keyVaultKey, TokenCredential credential, CryptographyClientOptions options) : this()
+        {
+            Check.NotNull(keyVaultKey, nameof(keyVaultKey));
+            Check.NotNull(options, nameof(options));
+            Check.NotNull(credential, nameof(credential));
+
+            KeyVaultKey = keyVaultKey;
+            KeyVaultKeyUri = KeyVaultKey.VaultUri;
+            CryptoClientOptions = options;
+            Credentials = credential;
+
+            BearerTokenAuthenticationPolicy bearerAuthPolicy = new BearerTokenAuthenticationPolicy(credential, DEFAULT_KV_SCOPE_URI);
+            Pipeline = HttpPipelineBuilder.Build(CryptoClientOptions, bufferResponse: true, clientPolicies: bearerAuthPolicy);
+
+            InitProvider();
+        }
+
+        #endregion
+
+        #endregion
+
+        #region Public Functions
+
+        #endregion
+
+        #region private functions
+        void InitProvider()
+        {
+            switch(KeyVaultKey.KeyMaterial.KeyType)
```

Comment:
it's possible that KeyVaultKey (which btw I would just call Key) could be null, if the user doesn't have the key_get permission.  This needs to be handled as well.

Reply:
Good catch, let me see and handle that scenario.